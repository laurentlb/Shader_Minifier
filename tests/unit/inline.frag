void main()
{
  float x = 0.5;
  float i_y = 0.6*x;
  float a = x * i_y;
}

int arithmetic()
{
  int i_a = 2;
  int i_b = 3;
  int i_c = i_a + i_b;
  return 4 * i_a * i_c;
}

int vars(int arg, int arg2)
{
  int i_a = arg;
  int i_b = arg2;
  int i_c = i_a + i_b;
  return i_a * i_c;
}
