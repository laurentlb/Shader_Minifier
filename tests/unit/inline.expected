#version 120

float result;
void main()
{
  result=.15;
}
int arithmetic()
{
  return 40;
}
int vars(int arg,int arg2)
{
  return arg*(arg+arg2);
}
int arithmetic2()
{
  return 40;
}
int unusedVars()
{
  return 30;
}
int unusedVars2()
{
  return 18;
}
int multiPass()
{
  return 3;
}
float multiPass2()
{
  return 9.;
}
atomic_uint hydrogen;
uint builtin_with_or_without_side_effects(uint x)
{
  uint not_inlined=atomicCounterIncrement(hydrogen);
  atomicCounterIncrement(hydrogen);
  return x+max(x*x,x+1)+not_inlined;
}
float dmin(float a,float b)
{
  return a+b+a*b;
}
float reduce_consecutive_assignments(float x)
{
  float dmat=dmin(dmin(dmin(dmin(dmin(34.+x,.2),.2),.3),.4),.2);
  dmat+=dmat*dmat;
  return dmat+x;
}
int dont_inline_lvalue()
{
  return 3;
}
vec4 fragColor247;
void main247()
{
  fragColor247=vec4(3);
}
vec4 fragColor248;
void main248()
{
  fragColor248=vec4(3);
}
float arr[]=float[](3.4,4.2);
void lvalues()
{
  arr[1]=2.;
}
uniform int time;
in int sync;
int dependOnConst()
{
  return(time+sync)*2*3;
}
float noinl179(float x)
{
  float old=x;
  x=1e2;
  return old+x;
}
float inlineWithShadowing(float x)
{
  float inl=sin(2.5);
  if(x<0.)
    {
      float inl=3.4;
      inl++;
    }
  return inl;
}
float inline_uninitialized()
{
  return c;
}
float glo;
float noinline_readsTheGlobal()
{
  return glo;
}
float dontCompressAssigments()
{
  glo=10.;
  glo=50.+noinline_readsTheGlobal();
  return glo*glo;
}
